hooks/query/use-task-history-websocket.ts

```diff
@@ -0,0 +1,72 @@
+import React from "react"
+
+import {
+  createSignedWebSocketRequest,
+  Task,
+  useWasmInitialized,
+} from "@renegade-fi/react"
+import { getSymmetricKey } from "@renegade-fi/react/actions"
+import { WS_TASK_HISTORY_ROUTE } from "@renegade-fi/react/constants"
+import useWebSocket, { ReadyState } from "react-use-websocket"
+
+import { useConfig, useCurrentWallet } from "@/providers/state-provider/hooks"
+
+import { parseBigJSON } from "./utils"
+
+export type UseTaskHistoryWebsocketParameters = {
+  onUpdate: (task: Task) => void
+}
+
+export function useTaskHistoryWebSocket(
+  parameters: UseTaskHistoryWebsocketParameters,
+) {
+  const { onUpdate } = parameters
+  const isWasmInitialized = useWasmInitialized()
+  const { id: walletId } = useCurrentWallet()
+  const config = useConfig()
+
+  const url = config?.getWebsocketBaseUrl() ?? ""
+  const enabled = Boolean(url && config)
```

Comment:
is this `enabled` state properly reactive to the selected wallet changing? presumably `currentWallet` changing would also trigger a change in `config`, but if that were sufficient we wouldn't have the need to port over these hooks, since they already relied on `useConfig`, right?

Reply:
yeah you got it, particularly the issue with the old hooks was that the config.state.seed and config.state.id were not reactive, leading to weird bugs. much easier to just have them be reactive and construct a new config anytime they change